### ISSUE TITLE
Backup and restore entire filesystem

### ISSUE THREAD

**Issue by onionhammer** [stackblitz/webcontainer-core#1113]
I'm really trying to minimize downtime between refreshes, so users can very quickly pickup where they left off, or barely even notice when they hit refresh on their browser.

Are there alternative ways to 'mount' files en-masse? Like pointing to a [browser cache](https://developer.mozilla.org/en-US/docs/Web/API/Cache) in order to skip the `npm install`?

**Describe the solution you'd like:**
It would be great to be able to have more utilities around `fs` for backing up and restoring/mounting data

**Comment by jrvidal** [comment]
:wave: hi @onionhammer, I assume you're talking about the `fs` exposed in the WebContainer API? 

"Mounting files en-masse" is probably your best bet. We don't expose any "backup" functionality and that is something we could explore. But I'll point you to learn.svelte.dev, where IIRC they don't actually run `npm install` at all. Instead they mount a full, pre-downloaded `node_modules` folder. Of course this only works b/c the dependency tree is static and known in advance, not sure if that would fit your use case.

**Comment by onionhammer** [comment]
Hmm yeah, even learn.svelte.dev takes maybe 8 seconds to show the preview. Ok thanks, maybe webcontainers arent a good fit for my scenario.

I was hoping to be able to use it in the 'live preview' of directus (the CMS), which displays your site in an iframe next to the editor. If the files could be 'npm installed' once by the user for every unique package-lock.json provided, then every time they update content, the live preview would reflect their CMS changes. When a developer updates the site layout itself, a github action would put up the new package-lock, along with the other source files of the static site, so that the webcontainer would run the site in "dev" mode, which talks to the CMS.

This scenario is obviously quite doable, but each time the CMS data changes and the user saves, it would probably take 6 or 7 seconds to actually reload the webcontainer, which is much slower than I have it working today

**Comment by jrvidal** [comment]
I probably don't fully grasp the finer details of your use case, but I'll just add that, after `npm install` has run and once the live preview is running, live reloading tends to be quite fast. You're correct that "booting" WebContainer and/or installing deps takes some time. But please note how [learn.svelte.dev](https://learn.svelte.dev) only needs to do that _once_, and from there you can use their editor/playground pretty seamlessly.

**Comment by onionhammer** [comment]
@jrvidal Yeah you're right, but that would require changes to how directus itself works in order to not just refresh the live preview iframe, but to trigger a hot-reload of the app running in the container instead.
